docs/UserGuide.md

```diff
@@ -65,81 +69,223 @@ AddressBook Level 3 (AB3) is a **desktop app for managing contacts, optimized fo
+* The index **must be a positive integer** 1, 2, 3, …​
+* Deleting a job already matched with a contact removes the match from the contact as well.
+
+Examples:
+* `list job` followed by `delete job 2` deletes the 2nd job in the address book.
+* If the job at index 2 is matched with contact at index 1, `delete job 2` will remove the 
+match from the contact at index 1.
+
+### Deleting a company : `delete company`
+
+Deletes the specified company from the address book.
+
+Format: `delete company INDEX`
+
+* Deletes the company at the specified `INDEX`.
+* The index refers to the index number shown in the displayed company list.
+* The index **must be a positive integer** 1, 2, 3, …​
+* Deleting a company with jobs attributed to it also causes those jobs to be deleted.
+
+Examples:
+* `list company` followed by `delete company 2` deletes the 2nd company in the address book.
+* If the company at index 1 has a job attributed to it, `delete company 1` will also delete the job.
+
+### Screening contacts by a job's name : `screen job`
+
+Screens the list of contacts in the address book with the name of the
+job specified.
+
+Format: `screen job INDEX`
+
+* Uses the name of the job at the specified `INDEX` to filter the 
+list of contacts to contacts with role matching the name.
+* The index refers to the index number shown in the displayed job list.
+* The index **must be a positive integer** 1, 2, 3, …​
+* The filter is case-insensitive i.e. `Cleaner` will match `cleaner`.
+
+Examples:
+* If the job at index 1 has name `Software Engineer`, `screen job 1` will
+show a contact with role `Software Engineer`.
+* If the job at index 2 has name `Data Scientist`, `screen job 2` will 
+show a contact with role `data scientist`.
+
+### Matching a contact with a job : `match`
+
+Matches a contact to a job.
+
+Format: `match CONTACT_INDEX JOB_INDEX`
+
+* Matches the contact at the specified `CONTACT_INDEX` and the job
+at the specified `JOB_INDEX` together.
+* The contact index refers to the index number shown in the displayed contact list.
+* The job index refers to the index number shown in the displayed job list.
+* Both indices **must be positive integers** 1, 2, 3, …​
+* Both the contact and job specified **cannot already be matched**.
+
+Examples:
+* `match 1 2` will match the contact at index 1 and job at index 2 together.
+* `match 2 3` will match the contact at index 2 and job at index 3 together.
+
+### Undoing a matched contact and job : `unmatch`
+
+Undoes a matching between a contact and job.
+
+Format: `unmatch CONTACT_INDEX JOB_INDEX`
+
+* Undoes the matching of a contact at the specified `CONTACT_INDEX` and a job
+at the specified `JOB_INDEX`.
+* The contact index refers to the index number shown in the displayed contact list.
+* The job index refers to the index number shown in the displayed job list.
+* Both indices **must be positive integers** 1, 2, 3, …​
+* The contact and job **must already be matched together**.
```

Comment:
This is fine, but as mentioned in the tele chat I will be making it such that job does not keep a reference anymore, so the last line in the match and unmatch might need to be amended to something like `Contact must not already by matched`.  

Earliest I will be able to do the change is on Monday, so apologies for that.

Reply:
Actually, maybe you could edit the UG after you change that to register that you edited the UG for this milestone.